openmc/checkvalue.py

```diff
@@ -1,3 +1,17 @@
+from collections import Iterable
+from numbers import Integral
+
+import numpy as np
+
+def _isinstance(value, expected_type):
+    """A Numpy-aware replacement for isinstance"""
+    np_ints = [np.intc, np.intp, np.int8, np.int16, np.int32, np.int64,
+               np.uint8, np.uint16, np.uint32, np.uint64]
+    if expected_type is Integral:
```

Comment:
Also, the NumPy issue applies to floats as well as integers. So you need to include a ``np_floats`` type with its own conditional block here.